foghorn_python/scripts/vcf_comp_funcs.py

```diff
@@ -0,0 +1,138 @@
+	#	lines_holder.append(string)
+		sample_index = sample_index + 1  
+		return sample_index, lines_holder
+	elif GT in {".|1", "1|.", "./1", "1/.", ".|0", "0|.", "./0", "0/."}: # missing variants, Key value with variants and sample is better than this. test dataset has little missing variants so not an issue just yet 
+		string = "".join(["Sample# ", str(sample_index), " has a single allele missing at pos ->  ", str(fields[:4])])
+		mis.append(string)
+		sample_index = sample_index + 1
+		return sample_index, mis
+	elif GT in { "./.", ".|."}:
+		string = "".join(["Sample# ", str(sample_index), " has a both alleles missing at pos ->  ", str(fields[:4])])
+		mis.append(string)
+		sample_index = sample_index + 1
+		return sample_index, mis
+	else:
+		sample_index = sample_index + 1
+		return sample_index
+
+def DS_comp(DS, lines_holder, sample_index):
+	if float(DS) != 0:
+		string= "".join([str(sample_index), ":", str(DS)])
+		lines_holder.append(string)
+		sample_index = sample_index + 1
+		return lines_holder, sample_index
+	else:
+		sample_index = sample_index + 1
+		return sample_index
+
+#def mult_DS(DS, lines_holderDS, sample_indexDS):
+#	if
+
+def mult_al(GT, lines_holderMS, sample_indexMS, mis):
+	if GT in {"2|2", "2/2"}:
+		string= "".join([str(sample_indexMS), ":", "2"])
+		lines_holderMS.append(string)
+		return sample_indexMS, lines_holderMS
+	elif GT in {"0|2", "2|0", "2/0", "0/2"}: # need to clarify the odds of seeing 1|2. This will most probably be a sequencing error. So for the moment I am only including the referernce "0"
+	# This is one implementation the second is to just add extra indices 
+		string= "".join([str(sample_indexMS), ":", "1"])
+		lines_holderMS.append(string)
+		return sample_indexMS, lines_holderMS
+	elif GT in {".|.","./."}:
+                string = "".join(["Sample# ", str(sample_index), " has a both alleles missing at pos ->  ", str(fields[:5])])
+                mis.append(string)
+                sample_indexMS = sample_indexMS + 1
+                return sample_indexMS, mis
+	elif GT in {".|2", "2|.", "2/.", "./2", ".|1", "1|.", "1/.", "./1",".|0", "0|.", "0/.", "./0"}:
+		string = "".join(["Sample# ", str(sample_index), " has asingle allele missing at pos ->  ", str(fields[:5])])
+                mis.append(string)
+                sample_indexMS = sample_indexMS + 1
+                return sample_indexMS, mis
+	else:
+		return None
+
+
+def GT(line, out, mis_vars):
+        if not line.startswith("##") and not line.startswith("#"):  # Need to spend time making robust and reporting file format and missingness errors
+                fields = line.strip().split()
+		if "GT" not in fields[8]:
+			sys.exit("No genotype tag in format field")# raise error that there is no GT fields in 
+                sample_index = sample_indexMS = 0
+                lines_holderMS = [] # refresh every iteration need to check if this has any effect on the memory management of large datasets
```

Comment:
The memory consumption should be fine here because the previous object has been dereferenced.